bin/varnishd/cache/cache_fetch.c

```diff
@@ -719,7 +725,11 @@ vbf_stp_error(struct worker *wrk, struct busyobj *bo)
 	// XXX: reset all beresp flags ?
 
 	HTTP_Setup(bo->beresp, bo->ws, bo->vsl, SLT_BerespMethod);
-	http_PutResponse(bo->beresp, "HTTP/1.1", 503, "Backend fetch failed");
+	if (bo->err_code > 0)
+		http_PutResponse(bo->beresp, "HTTP/1.1", bo->err_code, bo->err_reason);
+	else
+		http_PutResponse(bo->beresp, "HTTP/1.1", 503, "Backend fetch failed");
```

Comment:
It might be preferable to set `bo->err_{code,reason}` upon failure to avoid this branch.

Reply:
This would cause more code changes as everywhere we call the error state that isn't explicitly  from return error we would have to add these lines before it.

Reply:
@rezan  brought up a point saying if we change this in the struct then we lose the original user intent.

Reply:
Don't we have the handling for that?

On the same topic I would rather not execute `wrk->stats->fetch_failed++;` a few lines above if the aforementioned intent was to willingly go to `vcl_backend_error` and not a failed attempt to contact a backend.

Reply:
So you are saying if you use return error you don't want the fetch failed counter to be bumped?
11:41 AM is their a different counter that should be bumped instead?

Reply:
> So you are saying if you use return error you don't want the fetch failed counter to be bumped?

If I `return (error)` from either `vcl_backend_{fetch,response}` then there was either no fetch, or one succeeded. Short answer, yes.

> Is their a different counter that should be bumped instead?

Either none off the top of my head, or the other way around: we might want to have an equivalent to `MAIN.s_synth`.

Reply:
I would make the counter backend_error so people don't confused the counter being used for every error everywhere.

Reply:
I added a force push which makes the fetch_failed counter not increment when the call to ```vbf_stp_error``` was the result of a return(error) in VCL.